cEP-0019.md

```diff
@@ -0,0 +1,404 @@
+                      "totalCount": 1,
+                      "nodes": [
+                        {
+                          "content": "THUMBS_UP",
+                          "user": {
+                            "login": "li-boxuan"
+                          }
+                        }
+                      ]
+                    }
+                  }
+                ]
+              }
+            }
+          ]
+        },
+        "comments": {
+          "nodes": [
+            {
+              "bodyText": "To be honest, I don't understand why we need fetch-issues.js. From the relevant code, I can see fetch-issues.js does some pre-fetching and initial-data.js dumps it into the database. But what's the aim of doing that? It doesn't seem to save API hits. These pre-fetched issues in database are replaced later anyway.",
+              "author": {
+                "login": "li-boxuan",
+                "url": "https://github.com/li-boxuan"
+              },
+              "reactions": {
+                "totalCount": 0,
+                "nodes": []
+              }
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+The collection of meta-reviews includes two parts:
+
+##### Pre-fetch meta-reviews
+
+Currently, `fetch-issues.js` pre-fetches issues and pull requests to speed up
+gh-board performance and save users' API hits. Meta-reviews are also to be
+fetched during this procedure.
+
+##### Fetch meta-reviews
+
+When a new issue/pull request is detected, gh-board will fetch that issue and
+loads it asynchronously. Meta-reviews also need to be fetched during this
+procedure.
+
+### Analysis of Meta-reviews
+
+This section explains how meta-review information can be utilized to
+generate a meaningful analysis.
+
+As the collection of meta-reviews, this part also includes two parts:
+
+##### Pre-process meta-reviews
+
+After pre-fetching meta-reviews, they are to be processed, which is explained
+in the following sections.
+
+##### Process meta-reviews
+
+As new meta-reviews are loaded, useful information is to be extracted and
+current analysis of meta-review is to be updated correspondingly.
+
+The following sections explain how meta-review information is used to
+generate a meaningful analysis.
+
+#### Ranking list
+
+A ranking list would be displayed on coala community website, according
+to points people get. A sample formula for points would be as following:
+
+PR = total number of positive reactions (e.g., THUMBS_UP) a person gets
+for all reviews he did.
+NR = total number of negative reactions (e.g., THUMBS_DOWN) a person gets
+for all reviews he did.
+PG = total number of positive reactions a person gives to other people for
+their reviews.
+NG = total number of negative reactions a person gives to other people for
+their reviews.
+
+Note PG and NG are used to encourage people to get involved in this
+reactions feedback system. Also, it may not be limited to PRs. Then final
+score S would be:
```

Comment:
"Then final..." should be a new paragraph.

And how do we deal with cases where a maintainer provides a different meta-review than a developer?

One rather complicated way solve this is to weight meta-reviews based on the reviewers score.  Cyclical, but there are algorithms which provide a path towards equilibrium of the scores in the network.

Reply:
PageRank Algorithm is a strong candidate. However, my thought is that we want to encourage people to do meta-reviews by giving them a bonus.  Under this circumstance, algorithms that use Markov Chains like PageRank Algorithm don't fit well IMO. Maybe we could leave this for further discussion and enhancement. It would be quite interesting.

`weight meta-reviews based on the reviewers' score` is definitely a good idea and we should adopt that. The good news is we are calculating scores discretely: we only update scores when a new `issues.json` is fetched by coala/community repo. We could use the previous reviewers' score as a base and update the scores. This would not be cyclical anymore.